docs/planning/otel/OTelWebSdk.md

```diff
@@ -0,0 +1,1344 @@
+
+- **Interface-First Design**: All components implement well-defined TypeScript interfaces with no concrete dependencies
+- **Inversion of Control (IoC)**: Complete dependency injection pattern with no global state or singletons
+- **Factory-Based Instantiation**: Controlled SDK lifecycle through factory functions
+- **Resource Sharing**: Efficient coordination while maintaining instance isolation  
+- **OpenTelemetry Compliance**: Full adherence to OpenTelemetry specifications
+- **Explicit Dependency Management**: All dependencies injected through constructors, enabling testability and modularity
+
+For comprehensive architectural details including IoC principles, dependency injection patterns, component interactions, multi-instance factory design, and implementation guidelines, see **[OTelWebSdk-Architecture.md](./specs/OTelWebSdk-Architecture.md)**.
+
+### Anti-Patterns to Avoid
+
+#### **CRITICAL: Never Import OpenTelemetry Packages Directly**
+
+**The SDK MUST NOT import or depend on `@opentelemetry/*` packages** as many have side effects that automatically register global implementations, breaking the interface-first/factory function approach. These packages often cause:
+
+- **Global provider registration** that conflicts with multi-instance isolation
+- **Automatic instrumentation setup** that prevents controlled lifecycle management  
+- **Static configuration loading** that breaks dynamic configuration capabilities
+- **Side effect execution at module load time** that interferes with graceful degradation
+
+**Correct Approach:** The SDK must define its own **interface-only** contracts that are compatible with OpenTelemetry specifications through duck typing. This enables OpenTelemetry ecosystem compatibility while maintaining:
+
+- **Prevents Global State**: Avoids automatic global provider registration
+- **Enables Multi-Instance**: Multiple SDK instances can coexist without conflicts  
+- **Maintains Isolation**: Teams get truly isolated telemetry contexts
+- **Interface Compatibility**: Still works with OpenTelemetry ecosystem via duck typing
+- **Bundle Size**: Avoids pulling in unused OpenTelemetry infrastructure
+- **Tree-Shaking**: Ensures dead code elimination works properly
+
+### Key Implementation Principles
+
+The SDK implementation follows specific architectural patterns detailed in the [Architecture document](./specs/OTelWebSdk-Architecture.md):
+
+- **Interface-First Design**: TypeScript interfaces define all public contracts
+- **Factory Function Pattern**: Controlled instantiation with dependency injection  
+- **IoC Pattern**: No global state, explicit dependency management
+- **Closure-Based Implementation**: For bundle size optimization and true private state management
+- **High-Performance Architecture**: Minimal overhead design with advanced batching, resource management, and bundle optimization
+- **Modular Architecture**: Tree-shakable, plugin-based extensibility
```

Comment:
Will the new plugins mimic what we currently have (react, angular, etc.) or would expect new ones to be necessary?